mage/query-modules/python/xml_module.md

```diff
@@ -0,0 +1,614 @@
+```
+Output:
+
+```plaintext
+{
+   "_note": [
+      {
+         "_text": "Tove",
+         "_type": "to"
+      },
+      {
+         "_text": "Jani",
+         "_type": "from"
+      },
+      {
+         "_text": "Reminder",
+         "_type": "heading"
+      },
+      {
+         "_text": "Don't forget me this weekend!",
+         "_type": "body"
+      }
+   ],
+   "_type": "note"
+}
+```
+
+</TabItem>
+
+
+<TabItem value="usage3">
+
+Example of the file:
+
+```plaintext
+
+file.xml
+
+<catalog>
+  <book id="1">
+    <title>Book 1</title>
+    <author>Author 1</author>
+    <publication>
+      <year>2022</year>
+      <publisher>Publisher A</publisher>
+    </publication>
+  </book>
+  <book id="2">
+    <title>Book 2</title>
+    <author>Author 2</author>
+    <publication>
+      <year>2023</year>
+      <publisher>Publisher B</publisher>
+    </publication>
+  </book>
+</catalog>
+
+```
+
+Cypher:
+
+```cypher
+CALL xml_module.load("", true, "/home/demonstration/Documents/file.xml", "", {}) YIELD output_map RETURN output_map;
+```
+
+Output:
+
+```plaintext
+{
+   "_catalog": [
+      {
+         "_book": [
+            {
+               "_text": "Book 1",
+               "_type": "title"
+            },
+            {
+               "_text": "Author 1",
+               "_type": "author"
+            },
+            {
+               "_publication": [
+                  {
+                     "_text": "2022",
+                     "_type": "year"
+                  },
+                  {
+                     "_text": "Publisher A",
+                     "_type": "publisher"
+                  }
+               ],
+               "_type": "publication"
+            }
+         ],
+         "_type": "book",
+         "id": "1"
+      },
+      {
+         "_book": [
+            {
+               "_text": "Book 2",
+               "_type": "title"
+            },
+            {
+               "_text": "Author 2",
+               "_type": "author"
+            },
+            {
+               "_publication": [
+                  {
+                     "_text": "2023",
+                     "_type": "year"
+                  },
+                  {
+                     "_text": "Publisher B",
+                     "_type": "publisher"
+                  }
+               ],
+               "_type": "publication"
+            }
+         ],
+         "_type": "book",
+         "id": "2"
+      }
+   ],
+   "_type": "catalog"
+}
+```
+</TabItem>
+
+<TabItem value="usage4">
+
+For the XPath demonstration, [**this**](https://www.w3schools.com/xml/cd_catalog.xml) file will be used.
+
+The XPath expression is going to be `".//CD[YEAR='1988']"`, which will return all CD elements with attribute year equaling 1988. Note that XPath expressions cannot be absolute paths, because of the Python implementation of XPath, so `.` is used as an XPath prefix for this example, meaning the search is started from the current(root) element.
```

Comment:
```suggestion
The XPath expression is going to be `".//CD[YEAR='1988']"`, which will return all CD elements with attribute year equaling 1988. Note that XPath expressions cannot be absolute paths because of the Python implementation of XPath, so `.` is used as an XPath prefix for this example, meaning the search will start from the current(root) element.
```